src/Login/CertificationCheck/CertificationCheck.php

```diff
@@ -41,8 +41,12 @@ public function __construct(array $settings)
      *
      * @return bool
      */
-    public static function isXml($xml)
+    public static function isXml(string $xml)
     {
+        if ($xml[0] !== '<') {
+            return false;
+        }
+
```

Comment:
I don't get what you are trying to fix here? Please explain... The current logic yields the following result.

```shell
Psy Shell v0.8.18 (PHP 7.0.33-38+ubuntu20.04.1+deb.sury.org+1 — cli) by Justin Hileman
New version is available (current: v0.8.18, latest: v0.10.6)
>>> $input = "this is a test";
=> "this is a test"
>>> \Nodes\NemId\Login\CertificationCheck\CertificationCheck::isXml($input);
=> false
>>> $input = "<this is a test>";
=> "<this is a test>"
>>> \Nodes\NemId\Login\CertificationCheck\CertificationCheck::isXml($input);
=> false
>>> $input = "<note><to>Tove</to><from>Jani</from><heading>Reminder</heading><body>Don't forget me this weekend!</body></note>";
=> "<note><to>Tove</to><from>Jani</from><heading>Reminder</heading><body>Don't forget me this weekend!</body></note>"
>>> \Nodes\NemId\Login\CertificationCheck\CertificationCheck::isXml($input);
=> true
```

Reply:
When i was working on it, and the response was not XML (i.e. a normal error code) i would get a warning such as:
`Warning: DOMDocument::loadXML(): Start tag expected, '<' not found in Entity`

I cannot replicate your output, and actually get some strange behavior, where any non-xml input will make the function return true - likely because DOMDocument doesn't actually report any error, but if it was assigned to a variable which can be cast to boolean, the result would be as expected.
```
function isXml($xml) {
    try {
        $document = new \DOMDocument();
        $result = $document->loadXML($xml);

        return (bool) $result;
    } catch (\Exception $e) {
        return false;
    }    
}

isXml("this is a test"); // false
isXMl("<elm>hello</elm>"); // true
```

Reply:
Also if called with malformed XML such as `isXml("<elm>Hello<elm>")` then you'd get another warning:
`Warning:  DOMDocument::loadXML(): Premature end of data in tag elm line 1 in Entity`

I guess the only way to suppress the warnings, would be to set the error_reporting level, use the @-symbol such as `$result = @$document->loadXML($xml);`  or use a dummy error handler while trying to load the XML :(

Reply:
I am thinking perhaps we can use one the option constants instead the check you've added? 
https://www.php.net/manual/en/libxml.constants.php

Thinking of `LIBXML_NOWARNING`

```
function isXml($xml) {
    try {
        $document = new \DOMDocument();
        $result = $document->loadXML($xml, LIBXML_NOWARNING);

        return (bool) $result;
    } catch (\Exception $e) {
        return false;
    }    
}
```

Reply:
Using `LIBXML_NOWARNING` does not prevent the warnings for me, but `LIBXML_NOERROR` or `LIBXML_NOWARNING | LIBXML_NOERROR` does, but sure which would be the proper use, so i final version could be:

```php
function isXml($xml) {
    try {
        $document = new \DOMDocument();
        $result = $document->loadXML($xml, LIBXML_NOWARNING | LIBXML_NOERROR);

        return (bool) $result;
    } catch (\Exception $e) {
        return false;
    }
}
```
Not sure if  the try/catch is even needed, but I guess it doesn't hurt.

Reply:
Let's keep the try catch. I am not sure if the LIBXML_NOERROR prevents exceptions from being thrown. But let's get the PR updated with the flags that works for, and I will approve it. 👍

Reply:
I could test it, but i'm not really sure what would cause DOMDocument itself to throw an exception, but if it does then it would be caught, and the function return false.